browser/ui/views/split_view/split_view.cc

```diff
@@ -423,6 +453,66 @@ void SplitView::UpdateCornerRadius(const gfx::RoundedCornersF& corners) {
   secondary_devtools_web_view_->holder()->SetCornerRadii(corners);
 }
 
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+void SplitView::OnReaderModeToolbarActivate(ReaderModeToolbarView* toolbar) {
+  if (toolbar == secondary_reader_mode_toolbar_ &&
+      secondary_contents_web_view_) {
+    if (secondary_contents_web_view_->web_contents()->GetDelegate()) {
+      secondary_contents_web_view_->web_contents()
+          ->GetDelegate()
+          ->ActivateContents(secondary_contents_web_view_->web_contents());
+    }
+  }
+}
+
+void SplitView::UpdateSecondaryReaderModeToolbarVisibility() {
+  auto active_tab_handle = GetActiveTabHandle();
+  auto* split_view_browser_data =
+      SplitViewBrowserData::FromBrowser(base::to_address(browser_));
+  if (auto tile = split_view_browser_data->GetTile(active_tab_handle)) {
+    if (tile->first == active_tab_handle) {
+      secondary_reader_mode_toolbar_->SetVisible(IsTabDistilled(tile->second));
+    } else {
+      secondary_reader_mode_toolbar_->SetVisible(IsTabDistilled(tile->first));
+    }
+  } else if (secondary_reader_mode_toolbar_) {
+    secondary_reader_mode_toolbar_->SetVisible(false);
+  }
+}
+
+void SplitView::UpdateSecondaryReaderModeToolbar() {
+  auto* browser_view = static_cast<BraveBrowserView*>(browser_->window());
+  if (!browser_view) {
+    return;
+  }
+
+  UpdateSecondaryReaderModeToolbarVisibility();
+
+  auto* split_view_browser_data =
+      SplitViewBrowserData::FromBrowser(base::to_address(browser_));
+  if (split_view_browser_data &&
+      split_view_browser_data->IsTabTiled(GetActiveTabHandle())) {
+    // We need to swap the WebContents of the toolbars because, when the active
+    // browser tab is switched, the split view swaps both the views displaying
+    // the pages and the WebContents within those views. The toolbar does the
+    // same thing to ensure that the toolbar state follows the correct tab.
+    // DevTools views do the same.
+    ReaderModeToolbarView* primary_toolbar =
+        browser_view->reader_mode_toolbar();
+    primary_toolbar->SwapToolbarContents(secondary_reader_mode_toolbar_.get());
```

Comment:
Do we really need swap toolbar's content?
It seems toolbar functions for active tab's web contents. (Maybe I'm wronng).
Then, maybe we only need to activate web contents when toolbar gets clicked w/o this swap?

Reply:
Unfortuantely we need to do this.

Reply:
IMO, this seems fragile because whenever `UpdateSecondaryReaderModeToolbar()` is called, swap happens.
If it's called more time, I think it's possible to have invalid pair of toolbar and tab contents easily.
Should we have a way to make this swap happen when it's really necessary?